### ISSUE TITLE
Have `send_transaction` return the the onchain value if any

### ISSUE THREAD

**Issue by 0xmovses** [movementlabsxyz/movement#207]
In this function In thie function: https://github.com/movementlabsxyz/movement/blob/b70b559f3fa7d0fe8bb99e4ab80a730a6b1aa134/protocol-units/settlement/mcr/client/src/send_eth_transaction.rs#L61

It currently returns `Result<(), anyhow::Error>`, it would be greatly helpful for me in my bridge e2e testing if it could return the value on chain (if any) so `Result<Option<T>, anyhow::Error>`, where T is some solidity returned value. 

- Update the function to return some value. 
- Be sure not to break any of the functionality already implemented
- CI must pass.

**Comment by Icarus131** [comment]
@0xmovses, can this be done using `builder.call` with `alloy_contract::CallBuilder`?

**Comment by 0xmovses** [comment]
Yes indeed it is doable after fetching the receipt when calling `send()` on `call`. However, we are now just capturing the value we need via the event stream. So this is a stale issue now.
